In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('age-income.csv')
df

age  Income(k)
0   21         60
1   20         55
2   22         60
3   22         61
4   23         65
5   21         62
6   25         65
7   30         70
8   31         68

<h2>KNeighbors Regressor</h2>

In [4]:
from sklearn.neighbors import KNeighborsRegressor

In [5]:
neigh = KNeighborsRegressor(n_neighbors=3)

In [6]:
neigh.fit(df[['age']], df['Income(k)'])

KNeighborsRegressor(n_neighbors=3)

In [7]:
neigh.predict(df[['age']])

array([59.        , 59.        , 60.33333333, 60.33333333, 62.        ,
       59.        , 63.33333333, 67.66666667, 67.66666667])

In [12]:
neigh.predict([[22]])

C:\Users\hp\anaconda3\Lib\site-packages\sklearn\base.py:439: UserWarning: X does not have valid feature names, but KNeighborsRegressor was fitted with feature names
  warnings.warn(


array([60.33333333])

In [14]:
neigh.score(df[['age']], df['Income(k)'])

0.7401315789473687

In [15]:
y_pred1 = neigh.predict(df[['age']])
y_pred1

array([59.        , 59.        , 60.33333333, 60.33333333, 62.        ,
       59.        , 63.33333333, 67.66666667, 67.66666667])

In [16]:
df['y_pred1'] = neigh.predict(df[['age']])
df

age  Income(k)    y_pred1
0   21         60  59.000000
1   20         55  59.000000
2   22         60  60.333333
3   22         61  60.333333
4   23         65  62.000000
5   21         62  59.000000
6   25         65  63.333333
7   30         70  67.666667
8   31         68  67.666667

In [17]:
from sklearn.metrics import mean_squared_error

In [18]:
mse1 = mean_squared_error(df['Income(k)'], df['y_pred1'])
mse1

4.876543209876539

<h3>Linear Regression</h3>

In [19]:
from sklearn.linear_model import LinearRegression

In [24]:
df2 = pd.read_csv('age-income.csv')
df2

age  Income(k)
0   21         60
1   20         55
2   22         60
3   22         61
4   23         65
5   21         62
6   25         65
7   30         70
8   31         68

In [25]:
model1 = LinearRegression()
model1.fit(df2[['age']], df2['Income(k)'])

LinearRegression()

In [26]:
model1.predict([[22]])

C:\Users\hp\anaconda3\Lib\site-packages\sklearn\base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([60.97068966])

In [27]:
y_pred2 = model1.predict(df2[['age']])
y_pred2

array([59.95517241, 58.93965517, 60.97068966, 60.97068966, 61.9862069 ,
       59.95517241, 64.01724138, 69.09482759, 70.11034483])

In [28]:
df2['y_pred2'] = model1.predict(df2[['age']])
df2

age  Income(k)    y_pred2
0   21         60  59.955172
1   20         55  58.939655
2   22         60  60.970690
3   22         61  60.970690
4   23         65  61.986207
5   21         62  59.955172
6   25         65  64.017241
7   30         70  69.094828
8   31         68  70.110345

In [31]:
model1.score(df2[['age']], df2['Income(k)'])

0.7870258620689655

In [32]:
mse2 = mean_squared_error(df2['Income(k)'], df2['y_pred2'])
mse2

3.996551724137931

<h4 style='color:green'>If we apply KNN we got 74% accuracy. On the Other hand when we apply Linear Regression we got accuracy of 78%. If we compare these two algorithm for this dataset we got less Mean Squared Error in Linear Regression, having value of 3.99, but in KNN we got 4.7 </h4>